### ISSUE TITLE
feat: allow `Table.filter()` with no args, make it a noop

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#9720]
### Is your feature request related to a problem?

I have to have an  `if` guard to check if a `list[ibis.Deferred]` is empty before I can pass it into `.filter()`. See [here](https://github.com/NickCrews/ibis-widget/blob/main/src/ibis_widget/_widget.py#L90-L91) (PS, once this is a little more done I'm curious if you want to share or otherwise make [this extension](https://github.com/NickCrews/ibis-widget) more official, I really think it will improve my UX of using ibis)

### Describe the solution you'd like

```
def filter(self, *args, **kwargs):
    if not args and not kwargs:
        return self
    # continue
```

### What version of ibis are you running?

main

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
This seems like a footgun, since any mistakes in filter specification that result in empty arguments (like an incorrect list comprehension) will silently pass through and appear to have no effect.

We'd also want to make this behavior consistent throughout our APIs if did do it.

For example `order_by` must be called with at least one key, do we now want to allow that to do nothing if no keys are passed. What about `mutate`, should that also be no-op with nothing passed? `select`? What about `drop_null`/`fill_null`, `rename`, `relocate`? All of these need thinking through for such a change.

It's not as convenient for your particular case, but as the application/library developer you're the only person that knows whether an empty filter is reasonable thing to pass into Ibis from your API.

**Comment by gforsyth** [comment]
I'm going to close this out as unplanned.
That said, @NickCrews your `anywidget` thing looks awesome!

**Comment by NickCrews** [comment]
Hmmm good point, we would need to change all the other APIs too. I could be convinced that all those changes are good too. I think they're solvable, but this is definitely not a huge pain so let's leave for now. 

To play devils advocate for a moment though: I footgunned myself with the inverse: In my application code, at first I didn't do the "if not filters" and then I accidentally passed in no filters and I got an error. So both choices lead to footguns, just ones of different characteristics. I think I agree your example footgun is more likely than mine though.
